multiscale_run/utils.py

```diff
@@ -576,7 +579,28 @@ def get_subs_d(d: dict) -> dict:
     Returns:
     dict: A new dictionary containing only string key-value pairs.
     """
-    ans = {k: v for k, v in d.items() if isinstance(k, str) and isinstance(v, str)}
+
+    ans = (
+        {
+            k[2:-1]: v
+            for k, v in d.items()
+            if isinstance(k, str) and isinstance(v, str) and re.match(r"\$\{(.*?)\}", k)
+        }
+        | {
+            k[1:]: v
+            for k, v in d.items()
+            if isinstance(k, str) and isinstance(v, str) and re.match(r"\$(\w+)", k)
+        }
+        | {
+            k: v
+            for k, v in d.items()
+            if isinstance(k, str)
+            and isinstance(v, str)
+            and not re.match(r"\$(\w+)", k)
+            and not re.match(r"\$\{(.*?)\}", k)
+        }
+    )
```

Comment:
Seems like this could be simplified by just constructing one dictionary and performing optional substitutions on the key, something like this:
```
{
    re.sub(r"\$(?:\{(.*?)\}|(\w+))", r"\1", k): v
    ...
}
```

Reply:
done

Reply:
(nice! btw)